libqtile/backend/wayland/core.py

```diff
@@ -503,22 +514,39 @@ def _on_new_output(self, _listener: Listener, wlr_output: wlrOutput) -> None:
             y = box.y + box.height / 2
             self.warp_pointer(x, y)
 
-    def _on_output_layout_change(self, _listener: Listener, _data: Any) -> None:
+    def _on_output_layout_change(self, _listener: Listener = None, _data: Any = None) -> None:
         logger.debug("Signal: output_layout change_event")
         config = OutputConfigurationV1()
 
-        for output in self.outputs:
+        # disable mons that are no longer enabled
+        for output in self._outputs:
+            if output.wlr_output.enabled:
+                continue
+            head = OutputConfigurationHeadV1.create(config, output.wlr_output)
+            head.state.enabled = False
+            self.output_layout.remove(output.wlr_output)
+
+        for output in self._outputs:
+            if not output.wlr_output.enabled:
+                continue
+            # TODO: move this function to pywlroots
+            if (
+                lib.wlr_output_layout_get(self.output_layout._ptr, output.wlr_output._ptr)
```

Comment:
this seems like it will allocate/increment some kind of refcount? do we need to decrement that refcount to avoid leaks?

Reply:
I don't think this uses refcounting, the output layout is manually destroyed here: https://github.com/qtile/qtile/blob/ca7b28ef46721536d3e5b1ebf59b16d2ac93398b/libqtile/backend/wayland/core.py#L396

Reply:
I double checked and no we shouldn't free/release this. if you do that all sorts of funny stuff happens with the output layout. I also don't see that malloc or calloc is called in the wlroots source code, it just gives you the pointer to the existing output_layout_output